### PR TITLE
32/remove logic from repository

### PR DESCRIPTION
**Pull request by admy7**
Refactor dataset search, trying to separate responsibilities. Starting to set up a clean architecture in discovery service. 

New tests are coming in the next PR.

## Summary by Sourcery

Refactor the dataset search functionality by introducing a clean architecture approach. Separate the logic into distinct components for dataset ID collection, repository access, and facet building. Implement new interfaces and classes to support this architecture, and remove deprecated code and tests.

New Features:
- Introduce a new `SearchDatasetsQuery` use case to handle dataset search operations, integrating dataset ID collection and facet building.

Enhancements:
- Refactor the dataset search logic by separating responsibilities into distinct components, such as `DatasetIdsCollector`, `DatasetsRepository`, and `FacetsBuilder` interfaces.
- Implement `CkanDatasetIdsCollector` and `BeaconDatasetIdsCollector` to collect dataset IDs from CKAN and Beacon sources respectively.
- Add `CkanFacetsBuilder` and `BeaconFacetsBuilder` to build facet groups for dataset searches from CKAN and Beacon sources.

Tests:
- Remove outdated tests related to the old dataset repository and response mapper, reflecting the new architecture.

### CODE REVIEW

File: src/main/java/io/github/genomicdatainfrastructure/discovery/datasets/infra/ckan/persistence/CkanDatasetIdsCollector.java [hunk 1]
```diff
@@ -2,46 +2,53 @@
...
 
         var response = ckanQueryApi.packageSearch(
                 query.getQuery(),
                 facetsQuery,
-                query.getSort(),
-                query.getRows(),
-                query.getStart(),
-                SELECTED_FACETS,
+                null,
+                null,
```
Comment (admy7) [inline src/main/java/io/github/genomicdatainfrastructure/discovery/datasets/infra/ckan/persistence/CkanDatasetIdsCollector.java:38]: question: should rows be null here? By default, package search uses 10 rows.

Comment (brunopacheco1) [reply src/main/java/io/github/genomicdatainfrastructure/discovery/datasets/infra/ckan/persistence/CkanDatasetIdsCollector.java:38]: suggestion: you should check CKAN API to remove this limit. You could also check if it is possible to reduce the fields we retrieve. In t his context, we only need `identifier`.

File: src/main/java/io/github/genomicdatainfrastructure/discovery/datasets/infra/ckan/persistence/CkanFacetsBuilder.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Optional.ofNullable;
+
+@ApplicationScoped
+public class CkanFacetsBuilder implements FacetsBuilder {
+
+    private static final String CKAN_FACET_GROUP = "ckan";
+    private static final String SELECTED_FACETS = "[\"access_rights\",\"theme\",\"tags\",\"spatial_uri\",\"organization\",\"publisher_name\",\"res_format\"]";
```
[the remark this answers is not in the document]
Comment (brunopacheco1) [reply src/main/java/io/github/genomicdatainfrastructure/discovery/datasets/infra/ckan/persistence/CkanFacetsBuilder.java:24]: Selected facets have been moved to `application.properties` in this PR. 

https://github.com/GenomicDataInfrastructure/gdi-userportal-dataset-discovery-service/pull/93

**Review by inderps — APPROVED** [review]

**Comment by Markus92** [comment]
This broke test environment (http 409).

**Comment by zalborzi** [comment]
@Markus92 it will be reverted.

**Pull request merged**